internal/database/online/sqlutil/push.go

```diff
@@ -0,0 +1,32 @@
+package sqlutil
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/oom-ai/oomstore/internal/database/dbutil"
+	"github.com/oom-ai/oomstore/internal/database/online"
+	"github.com/oom-ai/oomstore/pkg/oomstore/types"
+)
+
+func BuildPushCondition(opt online.PushOpt, backend types.BackendType) (string, string, string, []interface{}, error) {
+	qt, err := dbutil.QuoteFn(backend)
+	if err != nil {
+		return "", "", "", nil, err
+	}
+
+	insertColumns := append([]string{opt.Entity.Name}, opt.FeatureNames...)
+	insertValues := append([]interface{}{opt.EntityKey}, opt.FeatureValues...)
+
+	updatePlaceholders := make([]string, 0, len(opt.FeatureNames))
+	for _, name := range opt.FeatureNames {
+		updatePlaceholders = append(updatePlaceholders, fmt.Sprintf("%s=?", qt(name)))
+	}
+
+	return qt(insertColumns...),
+		dbutil.Placeholders(len(insertColumns), "?", ","),
+		strings.Join(updatePlaceholders, ","),
+		append(insertValues, opt.FeatureValues...),
+		nil
+
```

Comment:
wrap return elements into a struct

Reply:
I'm also debating whether to wrap the return value. Let's change it for the next PR.